src/main/java/com/google/devtools/build/lib/analysis/constraints/TopLevelConstraintSemantics.java

```diff
@@ -274,7 +274,7 @@ public PlatformRestrictionsResult checkPlatformRestrictions(
         }
       }
     } catch (TargetCompatibilityCheckException e) {
-      throw new ViewCreationFailedException(e.getMessage(), e.failureDetail, e);
+      throw new ViewCreationFailedException(e.getMessage(), e.failureDetail);
```

Comment:
@katre has raised a good point here that it's better to keep the `Cause` since it would preserve the stack trace and lose the `e.getMessage()`: `throw new ViewCreationFailedException(e.getFailureDetail(), e)`

I would then modify [this constructor](https://github.com/bazelbuild/bazel/blob/09c621e4cf5b968f4c6cdf905ab142d5961f9ddc/src/main/java/com/google/devtools/build/lib/analysis/ViewCreationFailedException.java#L40) to instead call `super(cause.getMessage(), cause)`.

How does that sound?

Reply:
Yeah, that works for me :+1: 
Looks like you've made the appropriate changes already in https://bazel-review.googlesource.com/c/bazel/+/203153?
Do I need to update this PR to match?

Reply:
Actually, I will update this PR since I need to update the PR Summary as well.

Reply:
Updated the PR contents here and the PR Summary.